lib/datawarehouse/partitionRegistrar/index.ts

```diff
@@ -0,0 +1,87 @@
+        const { dataWarehouseBucket, athenaResultsBucket, scheduleExpression, table, partitionKey } = args;
+        const location = getS3Location(dataWarehouseBucket, table);
+
+        const options  = { parent: this }; 
+
+        const resultsBucket = athenaResultsBucket.arn.apply(a => `s3://${a.split(":::")[1]}`);
+
+        let lambdaAssumeRolePolicy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Action": "sts:AssumeRole",
+                    "Principal": {
+                        "Service": "lambda.amazonaws.com",
+                    },
+                    "Effect": "Allow",
+                    "Sid": "",
+                },
+            ],
+        };
+
+        const partitionRole = new aws.iam.Role(`${table}-partitionLambdaRole`, {
+            assumeRolePolicy: JSON.stringify(lambdaAssumeRolePolicy)
+        }, options);
+
+        const partitionGenLambdaAccess = new aws.iam.RolePolicyAttachment(`${table}-partition-lambda-access`, {
+            role: partitionRole,
+            policyArn: aws.iam.ManagedPolicies.AWSLambdaFullAccess
+        }, options);
+
+        const partitionGenAthenaAccess = new aws.iam.RolePolicyAttachment(`${table}-partition-athena-access`, {
+            role: partitionRole,
+            policyArn: aws.iam.ManagedPolicies.AmazonAthenaFullAccess
+        }, options);
+
+        const schedule = scheduleExpression ? scheduleExpression : `rate(1 hour)`;
+
+        const cron = new aws.cloudwatch.EventRule(`${table}-hourly-cron`, {
+            scheduleExpression: schedule
+        }, options);
+
+        cron.onEvent(`${table}-partitionregistrar`, new CallbackFunction(`${table}-partition-callback`, {
+            role: partitionRole,
+            callback: (event: EventRuleEvent) => {
+                // create an athena client here, write the 
+                const athena = require("athena-client");
+                const clientConfig = {
+                    bucketUri: resultsBucket.get()
+                };
+                const awsConfig = {
+                    region: args.region
+                };
+
+                const client = athena.createClient(clientConfig, awsConfig);
+
+                const query = createPartitionDDLStatement(args.database.name.get(), table, location.get(), partitionKey, event.time);
+
+                client.execute(query, (err: Error) => {
+                    if (err) {
+                        throw err;
+                    }
+                })
+            }
+        }), options);
+    }
+}
+
+export interface PartitionRegistrarArgs {
+    table: string;
+    partitionKey: string;
+    dataWarehouseBucket: aws.s3.Bucket;
+    athenaResultsBucket: aws.s3.Bucket;
+    database: aws.glue.CatalogDatabase;
+    region: string;
+    scheduleExpression?: string; // TODO: we should remove this. It's useful in active development, but users would probably never bother. 
```

Comment:
Do we see a case for not having hourly buckets? Maybe Daily or every minute?

If not, I can definitely see the benefit of taking this out.

Reply:
This was more about being able to set a `1 minute` interval during development when we're manually testing things. Just so we didn't have to wait an hour to see results between changes.